ui/StatusQ/src/plugin.cpp

```diff
@@ -57,6 +58,9 @@ class StatusQPlugin : public QQmlExtensionPlugin
         qmlRegisterType<WritableProxyModel>("StatusQ", 0, 1, "WritableProxyModel");
         qmlRegisterType<FormattedDoubleProperty>("StatusQ", 0, 1, "FormattedDoubleProperty");
 
+        qmlRegisterSingletonType<SharedUpdateTimer>(
```

Comment:
For me it looks a bit like exposing from c++ to qml what is already exposed.

Why not just sth like:

```
pragma Singleton

QtObject {
   signal triggered

   readonly property Timer d: Timer {
     interval: 1000
      ...
   }
}

```
It may do exactly the same, just using standard Timer.

Reply:
app state is also already exposed to qml and easy to check: https://doc.qt.io/qt-5/qml-qtqml-qt.html#application-prop, like `Qt.application.active`.

Reply:
Yup, done :) Just tested it, still works as expected